### PR TITLE
vPBX Users Location support

### PR DESCRIPTION
**Pull request by Kaian**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description
Adds a new entity that allows vPBX users to be grouped based on where they are working at.
Locations have no internal usage yet but they're pre-requisite to future implementations.

This is the upport for halliday of #1716

### CODE REVIEW

**Comment by ironArt3mis** [comment]
Rebase completed.

File: schema/DoctrineMigrations/Version20220322110232.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                          name VARCHAR(50) NOT NULL,
+                          description VARCHAR(500) DEFAULT NULL,
+                          companyId INT UNSIGNED NOT NULL,
+                      INDEX IDX_9517C8192480E723 (companyId),
+                      PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE Locations ADD CONSTRAINT FK_9517C8192480E723 FOREIGN KEY (companyId) REFERENCES Companies (id) ON DELETE CASCADE');
+
+        $this->addSql('ALTER TABLE Users ADD location INT UNSIGNED DEFAULT NULL');
+        $this->addSql('ALTER TABLE Users ADD CONSTRAINT FK_D5428AED5E9E89CB FOREIGN KEY (location) REFERENCES Locations (id) ON DELETE SET NULL');
+        $this->addSql('CREATE INDEX IDX_D5428AED5E9E89CB ON Users (location)');
```
Comment (mmadariaga) [inline schema/DoctrineMigrations/Version20220322110232.php:34]: Register this entity in PublicEntities in order to use API ACLs

File: web/client/src/entities/Location/Location.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    'description',
+];
+
+const location: EntityInterface = {
+    ...defaultEntityBehavior,
+    icon: ApartmentIcon,
+    iden: 'Location',
+    title: _('Location', { count: 2 }),
+    path: '/locations',
+    toStr: (row: any) => row.name,
```
Comment (mmadariaga) [inline web/client/src/entities/Location/Location.tsx:29]: toStr: (row: any) => row?.name || '' => would be better

**Review by mmadariaga — CHANGES_REQUESTED** [review]

**Comment by Kaian** [comment]
Updated with requested changes

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**